kythe/typescript/utf8.ts

```diff
@@ -28,24 +28,32 @@
 class Scanner {
   constructor(private buf: Buffer, public ofs = 0) {}
 
-  /** Scans forward one codepoint. */
-  scan(): void {
-    let byte = this.buf[this.ofs++];
+  /** Scans forward one codepoint and returns the delta in UTF-16 offset. */
+  scan(): number {
+    const byte = this.buf[this.ofs++];
 
     if ((byte & 0b10000000) === 0) {
       // Common case: ASCII.
-      return;
+      return 1;
     }
 
-    if ((byte & 0b11100000) == 0b11000000) {
```

Comment:
Please use named constants instead of binary literals.

Reply:
These come directly from the UTF-8 definition.  See the table at
  https://en.wikipedia.org/wiki/UTF-8#Description

So I'm not quite sure how to name them except by their value.  Any suggestions?

Reply:
That's a valid objection. Maybe leave a comment describing where they came from (with the link you'd posted)?